examples/nft-pawn-shop/index.rsh

```diff
@@ -30,15 +34,24 @@ export const main = Reach.App(() => {
   commit();
   
   B.interact.loanMade();
-  B.pay(loanAmt + interest)
-   .timeout(relativeTime(deadline), () => {
-      L.publish();
-      transfer(collateral).to(L);
-      commit();
-      exit();
-    });
-  transfer(collateral).to(B);
-  transfer(loanAmt + interest).to(L);
+
+  try {
+    const [[], k] = call(LA.payBack)
+      .pay(() => loanAmt + interest)
+      .assume(() => check(this == B))
+      .throwTimeout(relativeTime(deadline), null);
+    check(this == B);
+    k(null);
+    transfer(collateral).to(B);
+    transfer(loanAmt + interest).to(L);
+  } catch (_) {
+    // B timed out on paying back the loan
+    const [[], k] = call(LA.takeCollateral)
+      .assume(() => check(this == L));
```

Comment:
You actually don't need this... which is part of the reason you didn't include it in the consensus step. Anyone can give Elle the collateral... but probably only she will.

Reply:
It was an accidental exclusion. I have been wondering what is better practice, to be as lenient or strict as possible with triggering timeout actions. I just did it this way to stay on what seems intuitively like the safe side.

Reply:
I don't believe there is a strictness metric where you can decide to be more or less strict. There's just the right thing for the application... I think about it in terms of benefits. Like...

1. If X acts and then action B happens, which decides whether X "won" or not, then B must not be done by X, so we want to constrain it so someone else does it.
2. If action C spends resources of Y, then Y must be the only one that does it.
3. If action D returns resources to Z, then anyone can do it, but probably only Z will.

I think this program is like situation 3.